sync/class.jetpack-sync-module-plugins.php

```diff
@@ -30,6 +31,14 @@ public function init_before_send() {
 	}
 
 	public function check_upgrader( $upgrader, $details) {
+		if (
+			isset( $details['type'] ) &&
+			'plugin' == $details['type'] &&
+			is_wp_error( $upgrader->skin->result )
+		) {
+			do_action( 'jetpack_plugin_update_failed', $details );
```

Comment:
I really like this approach. 
Should we also send the error info. Like the error code and the error message? 

Can we write a test for this somehow?

Reply:
Plus 1 to Enej's comment